app/controllers/membership_applications_controller.rb

```diff
@@ -1,18 +1,18 @@
 class MembershipApplicationsController < ApplicationController
 before_action :get_membership_application, only: [:show, :edit, :update]
 before_action :authorize_membership_application, only: [ :update, :show, :edit]
-
   def new
     @membership_application = MembershipApplication.new
   end
 
   def create
     @membership_application = current_user.membership_applications.new(membership_application_params)
-    if @membership_application.save
+    if @membership_application.save && Orgnummer.new(@membership_application.company_number).valid?
```

Comment:
Should reverse the order of these 2 conditional checks. The `Orgnummer...` validation is cheapest/quickest.  If it is going to fail, we don't want to have to also spend the time on doing a `.save`.